fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBEnvironment.java

```diff
@@ -578,8 +584,26 @@ public List<Long> getDatabaseNames() {
                 continue;
             }
 
-            long db = Long.parseLong(name);
-            ret.add(db);
+            if (Strings.isNullOrEmpty(prefix)) { // default GlobalStateMgr db
+                if (StringUtils.isNumeric(name)) {
+                    long db = Long.parseLong(name);
+                    ret.add(db);
+                } else {
+                    // skip non default GlobalStateMgr db
+                }
+            } else {
+                if (name.startsWith(prefix)) {
+                    String dbStr = name.substring(prefix.length());
```

Comment:
If db full name is `starmgr123456`,  `dbStr` is `3456` if prefix is `starmgr12`, and is `123456` if prefix is `starmgr`.
How to distinguish this case?

Reply:
We only have two prefixes. one is "" and the other is starmgr. It is not a universal way

Reply:
empty or end with '_'